lib/src/screens/home_screen/home_screen.dart

```diff
@@ -35,42 +48,73 @@ class HomeScreen extends StatelessWidget {
               ))
         ],
       ),
-      body: Container(
-          decoration:
-              const BoxDecoration(gradient: ColorConstants.backgroundGradient),
-          child: Column(
-            children: [
-              Expanded(
-                flex: 1,
-                child: Container(),
-              ),
-              Expanded(
-                flex: 9,
-                child: ListView.builder(
-                  padding: const EdgeInsets.only(top: 20),
-                  itemCount: 10,
-                  itemBuilder: (context, index) {
-                    return Padding(
-                        padding: const EdgeInsets.only(
-                            left: 10, right: 10, bottom: 5),
-                        child: CustomCard(
-                          index: index,
-                          onTap: () {
-                            Navigator.of(context)
-                                .pushNamed(NameRoutesConstants.detailScreen);
+      body: BlocBuilder<CoinBloc, CoinState>(
+        builder: (context, state) {
+          if (state is CoinLoadInProgress) {
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          }
+          if (state is CoinLoadFailure) {
+            return Container(
+              color: Colors.red,
+              alignment: Alignment.center,
+              child: Text(state.errorMessage!),
+            );
+          }
+          if (state is CoinLoadSucess) {
+            var coin = state.coins;
```

Comment:
why is the variable named `coin` when the state property is `coins`?

Reply:
I will change the named